spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -102,6 +112,17 @@ public UnicastSendingMessageHandler(String host, int port) {
 	}
 
 	/**
+	 * Basic constructor using external datagram socket for sending outgoing data.
+	 * @param socketExpression socket expression describing external socket
+	 */
+	public UnicastSendingMessageHandler(String socketExpression) {
+		super("", 0);
+		this.mapper.setLengthCheck(false);
+		this.mapper.setAcknowledge(false);
+		this.socketExpression = SPEL_EXPRESSION_PARSER.parseExpression(socketExpression);
```

Comment:
`Assert.hasText(socketExpression)` must be here.

In addition we should have on more ctor for the `Expression` variant to have more flexibility in case of Java Config.